base/array.jl

```diff
@@ -2162,35 +2162,35 @@ indmin(a) = findmin(a)[2]
 """
     indexin(a, b)
 
-Return a vector containing the highest index in `b` for
-each value in `a` that is a member of `b` . The output
-vector contains 0 wherever `a` is not a member of `b`.
+Return an array containing the highest index in `b` for
+each value in `a` that is a member of `b`. The output
+array contains `nothing` wherever `a` is not a member of `b`.
 
 # Examples
 ```jldoctest
-julia> a = ['a', 'b', 'c', 'b', 'd', 'a'];
+julia> a = ['a', 'b', 'c', 'b', 'd', 'a']
 
-julia> b = ['a','b','c'];
+julia> b = ['a', 'b', 'c']
 
-julia> indexin(a,b)
+julia> indexin(a, b)
 6-element Array{Int64,1}:
  1
  2
  3
  2
- 0
+  nothing
  1
 
-julia> indexin(b,a)
+julia> indexin(b, a)
 3-element Array{Int64,1}:
  6
  4
  3
 ```
 """
-function indexin(a::AbstractArray, b::AbstractArray)
-    bdict = Dict(zip(b, 1:length(b)))
-    [get(bdict, i, 0) for i in a]
+function indexin(a, b::AbstractArray)
+    bdict = Dict(zip(b, eachindex(b)))
```

Comment:
Should use `keys(b)`.

Reply:
Or even `_pairs`, see my comment above.

Reply:
`pairs` returns index=>value pairs; this needs them the other way around.

Reply:
Ah, yes. I guess we could define `_keys` similar to `_pairs`, but for now using `keys` should be enough, we can always support non-`AbstractVector` arguments later.